plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -152,6 +152,14 @@ private void validateCatalogs(List<CatalogMetadata> catalogs) {
             "Missing Name Field from a catalog. Name is a required parameter.");
       }
 
+      if (!catalog.getName().matches(CATALOG_NAME_REGEX)) {
```

Comment:
As I understand, this is more like a post-verification. If user configured an "illegal" catalog name, where this exception will be captured? Is the exception blocking SQL plugin bootstrap?

I'm wondering the reason of adding this restriction. Because the regex pattern defined here is also inconsistent with ANTLR grammar.

Reply:
The restriction is to not include `.` in catalog name which is what we are relying to separate [catalogname, tablefunction] or [catalogname, tablename].

Steps to configure catalogs.
1) Add catalog.json in key-store using bin/opensearch-keystore command.
2) Bootstrap OpenSearch and Plugins.[Illegal catalog names block SQL Plugin bootstrap] 
Since the user is trying to setup catalogs, I have made a decision to block the plugin bootstrap. Let me know if you have other thoughts. 

Ignore through error messages?

Yes, regex pattern defined is inconsistent with ANTLR grammar.

```
qualifiedName
    : ident (DOT ident)*                             #identsAsQualifiedName
    ;

ident
    : (DOT)? ID
    | BACKTICK ident BACKTICK
    | BQUOTA_STRING
    | keywordsCanBeId
    ;
    
fragment ID_LITERAL:                [@*A-Z]+?[*A-Z_\-0-9]*;    
```


@dai-chen  One more question, can we determine catalogname, tablename in AST layer itself. We are loosing information of whether it is a single BQUOTA String or multiple identifiers with catalogname and tablename after AST layer. Will also check if we can pass this info till Analyzer layer by passing list of identifiers instead of tablename.

will change the regex to sync with ID_LITERAL [@*A-Z]+?[*A-Z_\-0-9]*